core/lib/utils/app_logger.dart

```diff
@@ -3,17 +3,38 @@ import 'dart:async';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 
-final logHistory = _Dispatcher("");
+final logHistory = _Dispatcher('');
 
-void log(String? value) {
-  if (kDebugMode) {
-    String v = value ?? "";
-    logHistory.value = "$v\n${logHistory.value}";
-    print(v);
+void log(String? value, {Level level = Level.info}) {
+  if (!kDebugMode) return;
+
+  String logsStr = value ?? '';
+  logHistory.value = '$logsStr\n${logHistory.value}';
+
+  switch (level) {
+    case Level.wtf:
+      logsStr = '\x1B[31m!!!CRITICAL!!! $logsStr\x1B[0m';
+      break;
+    case Level.error:
+      logsStr = '\x1B[31m$logsStr\x1B[0m';
```

Comment:
But how developer know it's an error, a warning, ...

Reply:
IMO, It's clearly marked by color, enum value and `TeamMail` key.